community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/OnlineIndexUpdatesValidator.java

```diff
@@ -60,25 +62,34 @@ public OnlineIndexUpdatesValidator( NeoStores neoStore, DatabaseHealth databaseH
     @Override
     public ValidatedIndexUpdates validate( TransactionRepresentation transaction ) throws IOException
     {
-        NodePropertyCommandsExtractor extractor = new NodePropertyCommandsExtractor();
-        try
+        // NodePropertyCommandsExtractor doesn't actually do anything in close atm, but wrap in try-with-resources
+        // in case it will in the future
+        try ( NodePropertyCommandsExtractor extractor = new NodePropertyCommandsExtractor() )
         {
-            transaction.accept( extractor );
+            try ( TransactionApplier txApplier = extractor.startTx( new TransactionToApply( transaction ) ))
+            {
+                transaction.accept( txApplier );
+            }
+            catch ( IOException cause )
+            {
+                databaseHealth.panic( cause );
+                throw cause;
+            }
+
+            if ( !extractor.containsAnyNodeOrPropertyUpdate() )
+            {
+                return ValidatedIndexUpdates.NONE;
+            }
+
+            Iterable<NodePropertyUpdate> updates = new LazyIndexUpdates( nodeStore, propertyStore, propertyLoader,
+                    extractor.propertyCommandsByNodeIds, extractor.nodeCommandsById );
+
+            return indexing.validate( updates, updateMode );
         }
-        catch ( IOException cause )
+        catch ( Exception cause )
```

Comment:
Because some implementation of CommandVisitor (I think..) throws something other than IOException.